### PR TITLE
Fix errors, warnings and deprecations from Qt 5.15

### LINKED ISSUE #1370: Build error with qt-5.15.0 - src/graphics/bitmap/bitmapimage.cpp:615:41: error: ‘path’ has incomplete type
Build against newly released qt-5.15.0 fails.

> x86_64-pc-linux-gnu-g++ -c -include .obj/core_lib -O2 -pipe -march=native -std=c++11 -fPIC -std=gnu++11 -D_REENTRANT -Wall -Wextra -DAPP_VERSION=\"0.6.4\" -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050600 -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_GUI_LIB -DQT_XML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I. -Isrc -Isrc/graphics -Isrc/graphics/bitmap -Isrc/graphics/vector -Isrc/interface -Isrc/structure -Isrc/tool -Isrc/util -Isrc/managers -Isrc/external -Isrc/external/linux -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtSvg -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtMultimedia -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtCore -I.moc -isystem /usr/include/libdrm -I.ui -I/usr/lib64/qt5/mkspecs/linux-g++ -o .obj/bitmapimage.o src/graphics/bitmap/bitmapimage.cpp
src/graphics/bitmap/bitmapimage.cpp:615:41: error: ‘path’ has incomplete type
  615 | void BitmapImage::drawPath(QPainterPath path, QPen pen, QBrush brush,
      |                            ~~~~~~~~~~~~~^~~~
In file included from /usr/include/qt5/QtGui/qtransform.h:43,
                 from /usr/include/qt5/QtGui/qimage.h:48,
                 from /usr/include/qt5/QtGui/qpixmap.h:49,
                 from /usr/include/qt5/QtGui/qpainter.h:48,
                 from /usr/include/qt5/QtGui/QPainter:1,
                 from src/graphics/bitmap/bitmapimage.h:21,
                 from src/graphics/bitmap/bitmapimage.cpp:17:
/usr/include/qt5/QtGui/qmatrix.h:54:7: note: forward declaration of ‘class QPainterPath’
   54 | class QPainterPath;
      |       ^~~~~~~~~~~~
make[1]: *** [Makefile:1494: .obj/bitmapimage.o] Error 1

### PR DESCRIPTION
**Pull request by J5lx**
With the recent release of Qt 5.15, I once again had a closer look at my compiler output. Most of the changes are rather boring, but it’d be nice if someone could have a look at the copy assignment operators. I based them on existing, related code, but since I’m not exactly familiar with that part of the codebase I’m not really sure if that is sufficient.

Fixes #1370.

### CODE REVIEW

**Comment by J5lx** [comment]
I also just saw that one of the issues I fixed here has been already been reported, so this PR will actually fix #1370 as well.

File: core_lib/src/managers/playbackmanager.cpp [hunk 1]
```diff
@@ -111,20 +111,6 @@ void PlaybackManager::play()
...
-        for (int i = 0; i < object()->getLayerCount(); ++i)
-        {
-            Layer* layer = object()->getLayer(i);
-            if (layer->type() == Layer::SOUND)
-            {
-                key = layer->getKeyFrameWhichCovers(frame);
-            }
-        }
-    }
-
```
Comment (chchwy) [inline core_lib/src/managers/playbackmanager.cpp:127]: Haven't checkout the branch with an IDE just quickly glanced through the code. Is there any reason to remove this section?

Comment (J5lx) [reply core_lib/src/managers/playbackmanager.cpp:127]: No, but there isn’t any reason to keep it either as far as I can see. The value that’s computed here is never used afterwards.

Comment (chchwy) [reply core_lib/src/managers/playbackmanager.cpp:127]: fair enough

**Pull request merged**